assets/src/components/form/inputs/checkbox-input.js

```diff
@@ -12,110 +12,93 @@ import { OptionInputs } from './base/option-inputs';
 import { OptionLabelSize } from './base/option-label-size';
 
 /**
- * generates one or more html checkbox inputs
+ * Generates one or more html checkbox inputs
  *
- * @constructor
- * @param {string} htmlId
- * @param {string} htmlClass
- * @param {string} value
- * @param {Array} options
- * @param {Function} onChange
- * @param {boolean} btnGroup
- * @param {string} helpTextID
- * @param {Object} dataSet
- * @param {string} inputWidth
- * @param {Object} attributes
- * @return {string} rendered date name form row
+ * @function
+ * @param {Object} props
+ * @return {string} rendered checkbox inputs
  */
-export class CheckboxInput extends Component {
-	static propTypes = {
-		htmlId: PropTypes.string.isRequired,
-		htmlClass: PropTypes.string,
-		value: PropTypes.oneOfType( [
-			PropTypes.array,
-			PropTypes.number,
-			PropTypes.string,
-		] ),
-		options: PropTypes.array,
-		onChange: PropTypes.func,
-		btnGroup: PropTypes.bool,
-		inputWidth: PropTypes.string,
-		helpTextID: PropTypes.string,
-		dataSet: PropTypes.object,
-	};
+export const CheckboxInput = ( {
+	name,
+	htmlId,
+	options,
+	required,
+	helpTextID,
+	btnGroup = true,
+	inputWidth = '',
+	...attributes
+} ) => {
+	const [ checked, setChecked ] = useState( resetState() );
```

Comment:
so `resetState` is being hoisted above this usage... 
but isn't that **just** the declaration?
meaning the memory space is reserved for something named `resetState` but the actual function doesn't get assigned to that name until we reach that in code?
or do function declarations get hoisted immediately?

Reply:
Unlike variables, a function declaration doesn't just hoist the function's name. It also hoists the actual function definition.

Reply:
ahh kk cool... that makes sense then